test/y2network/clients/routing_test.rb

```diff
@@ -34,5 +58,157 @@
         subject.main
       end
     end
+
+    context "when calling with 'list'" do
+      let(:args) { ["list"] }
+      it "prints the list of the configured routes" do
+        expect(Yast::CommandLine).to receive(:Print).with(/Routing Table/)
+        subject.main
+      end
+    end
+
+    context "when calling with 'show'" do
+      let(:args) { ["show"] }
+
+      context "and do not specify a dest target" do
+        it "prints and error" do
+          expect(Yast::CommandLine).to receive(:Print).with(/No entry for/)
+          subject.main
+        end
+      end
+
+      context "with an existent route destination" do
+        let(:args) { ["show", "dest=default"] }
+
+        it "prints the routes for the given destination" do
+          expect(Yast::CommandLine).to receive(:Print).with(/default[\s]+192.168.1.1[\s]+-[\s]+-/)
+          subject.main
+        end
+      end
+    end
+
+    context "when calling with 'add'" do
+      let(:args) { ["add"] }
+
+      context "and do not specify a dest target" do
+        it "prints and error" do
+          expect(Yast::CommandLine).to receive(:Print).with(/At least destination/)
+          subject.main
+        end
+      end
+
+      context "with at least a route destination" do
+        let(:args) { ["add", "dest=192.168.1.0", "gateway=192.168.1.1", "netmask=255.255.255.0"] }
+
+        it "adds the new route" do
+          expect(routing.routes.size).to eq(1)
+          subject.main
+          expect(routing.routes.size).to eq(2)
+        end
+      end
+    end
+    context "when calling with 'delete'" do
+      let(:args) { ["delete"] }
+
+      context "and do not specify a dest target" do
+        it "prints and error" do
+          expect(Yast::CommandLine).to receive(:Print).with(/No entry for/)
+          subject.main
+        end
+      end
+
+      context "with an existent route destination" do
+        let(:args) { ["delete", "dest=default"] }
+
+        it "deletes the routes with the given destination" do
+          expect(routing.routes.size).to eq(1)
```

Comment:
This "expect" call is not necessary as you already know the value. If you want to express the change, you can use the `expect {}.change {}.from().to()` form. Check below.